openeo/extra/spectral_indices/spectral_indices.py

```diff
@@ -0,0 +1,99 @@
+from openeo.processes import ProcessBuilder, array_modify
+from openeo.rest.datacube import DataCube
+import numpy as np
+
+def _get_expression_map(cube: DataCube, x: ProcessBuilder):
+    collection_id = cube.graph["loadcollection1"]["arguments"]["id"]
+    bands = [band.replace("0", "").upper() for band in cube.metadata.band_names]
+
+    def check_validity():
+        if not all(band in orig_bands for band in bands):
+            raise ValueError(
+                "The bands in your cube {} are not all Sentinel-2 bands (the following are: {})".format(bands,orig_bands))
+
+    def get_params():
+        lookup = dict(zip(orig_bands, norm_bands))
+        return {lookup[key]: x.array_element(i) for i, key in enumerate(bands)}
+
+    def lookupS2():
+        orig_bands = ["B1", "B2", "B3", "B4", "B5", "B6", "B7", "B8", "B8A", "B9", "B11", "B12"]
+        norm_bands = ["A", "B", "G", "R", "RE1", "RE2", "RE3", "N", "RE4", "WV", "S1", "S2"]
+        return orig_bands, norm_bands
+
+    def lookupL8():
+        orig_bands = ["B1", "B2", "B3", "B4", "B5", "B6", "B7", "B10", "B11"]
+        norm_bands = ["A", "B", "G", "R", "N", "S1", "S2", "T1", "T2"]
+        return orig_bands, norm_bands
+
+    def lookupL457():
+        orig_bands = ["B1", "B2", "B3", "B4", "B5", "B6", "B7"]
+        norm_bands = ["B", "G", "R", "N", "S1", "T1", "S2"]
+        return orig_bands, norm_bands
+
+    def lookupModis():
+        orig_bands = ["B3", "B4", "B1", "B2", "B5", "B6", "B7"]
+        norm_bands = ["B", "G", "R", "N", np.nan, "S1", "S2"]
+        return orig_bands, norm_bands
+
+    def lookupProbav():
+        orig_bands = ["BLUE", "RED", "NIR", "SWIR"]
+        norm_bands = ["B", "R", "N", "S1"]
+        return orig_bands, norm_bands
+
+    lookupPlatform = {
+        "LANDSAT4-5_TM_L1": lookupL457,
+        "LANDSAT4-5_TM_L2": lookupL457,
+        "LANDSAT8_L1": lookupL8,
+        "LANDSAT8_L2": lookupL8,
+        "MODIS": lookupModis,
+        "PROBAV_L3_S5_TOC_100M": lookupProbav,
+        "PROBAV_L3_S10_TOC_333M": lookupProbav,
+        "SENTINEL2_L1C_SENTINELHUB": lookupS2,
+        "SENTINEL2_L2A_SENTINELHUB": lookupS2,
+        "TERRASCOPE_S2_TOC_V2": lookupS2,
```

Comment:
I think we should avoid hardcoding names like SENTINELHUB and TERRASCOPE here

Reply:
I think the point is about just to detect if a collection is  Sentinel1, Sentinel2, landsat, modis, ...? The exact collection_id is not that relevant.

In most cases you can guess this pretty well from the collection name (e.g. `"S2" in collection_id or "SENTINEL2" in collection_id`),
but even better would be using explicit collection metadata like "constellation" or "platform"

Reply:
All these lookups seem mostly unnecessary if we'd properly fill the metadata with the common names as defined by STAC and the indices repo. They look very similar and then you could simply read the bands from the metadata without any hardcoding, I think.

Reply:
this is about building the mapping from  band names from collection metadata to the names used in the formulas in that "spectral indices json file" (which is bit different from common names), so you need some degree of hardcoding I'm afraid

Reply:
Hmm, as far as I understoof https://github.com/davemlz/awesome-spectral-indices#expressions all expressions consist of the "Standard" values, which seem to be somewhat aligned with https://github.com/stac-extensions/eo#common-band-names - I think the common names would make this work (more or less) automatically, similarly to the NDVI process in openEO, which implements the same idea.